tests/rp/src/bin/gpio_async.rs

```diff
@@ -0,0 +1,38 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use defmt::{assert, *};
+use embassy::executor::Spawner;
+use embassy_rp::gpio::Flex;
+use embassy_rp::Peripherals;
+use {defmt_rtt as _, panic_probe as _};
+
+#[embassy::main]
+async fn main(_spawner: Spawner, p: Peripherals) {
+    info!("embassy-rp gpio_async test");
+
+    let mut pin = p.PIN_16;
+    let mut async_pin = Flex::new(&mut pin);
+
+    // Test wait_for_high
+    {
+        async_pin.set_high();
+        async_pin.set_as_output();
+        delay();
+        assert!(async_pin.is_high(), "async_pin was expected to be high");
+
+        async_pin.set_as_input();
+        async_pin.wait_for_high().await;
```

Comment:
The tests are run automatically in CI, they should work without a human interacting with the board. 

The board in CI has GPIO0 and GPIO1 connected together, this is how the `gpio.rs` tests work (one pino utputs something, the other pin looks that it output the right thing).

Perhaps here the best way to test it is to set GPIO0 low, start a `wait_for_high` on GPIO1, check it doesn't return, then set GPIO0 high, check that it returns.

Reply:
Thanks for the details about how this works!

>Perhaps here the best way to test it is to set GPIO0 low, start a wait_for_high on GPIO1, check it doesn't return, then set GPIO0 high, check that it returns.

I've added a suggestion in https://github.com/embassy-rs/embassy/pull/853/commits/b29865b0698baa21110952b254479a77d233747c. Is something like that what you had in mind when you wrote `check it doesn't return`, or is there perhaps a way to check the future would return Poll::Pending which I'm not aware (or some other way that I'm just not "seeing")?

Reply:
@Dirbaio Just a friendly ping to see if you have time to take a look at my comment above? Thanks

Reply:
perhaps timing the `wait_for_high` and check it takes the expected time would work:

```rust
output.set_low();
let fut1 = async {
  Timer::after(Duration::from_millis(10)).await;
  output.set_high();
};
let fut2 = async {
  let start = Instant::now();
  input.wait_for_high().await;
  let dur = Instant::now() - start;
  assert!(dur >= Duration::from_millis(10) && dur < Duration::from_millis(11));
};
join(fut1, fut2).await;

Reply:
Oh, very nice! Thank you for showing me that!
I've updated this PR with a commit now.